mongo/gridfs/bucket.go

```diff
@@ -465,33 +506,31 @@ func createIndexIfNotExists(ctx context.Context, iv mongo.IndexView, model mongo
 		_ = c.Close(ctx)
 	}()
 
-	var found bool
+	modelKeysBytes, err := bson.Marshal(model.Keys)
+	if err != nil {
+		return err
+	}
+	modelKeysDoc := bsoncore.Document(modelKeysBytes)
+
 	for c.Next(ctx) {
 		keyElem, err := c.Current.LookupErr("key")
 		if err != nil {
 			return err
 		}
 
-		keyElemDoc := keyElem.Document()
-		modelKeysDoc, err := bson.Marshal(model.Keys)
-		if err != nil {
-			return err
-		}
+		keyElemDoc := bsoncore.Value{Type: keyElem.Type, Data: keyElem.Value}.Document()
```

Comment:
You can do `keyElem.Document()` here and cast to `bsoncore.Document` on the line where you call `indexDocsEqual`.

Reply:
done